api/src/routes/ranks.ts

```diff
@@ -0,0 +1,287 @@
+/**
+ * @swagger
+ * tags:
+ *   name: Ranks
+ *   description: Rank Management
+ */
+
+import express, { Request, Response } from "express";
+import logger from "../utilities/logger";
+import pool from "../database/index";
+import { Rank } from "../utilities/types";
+import { indexOf } from "lodash";
+import { paginationIsValid } from "../utilities/validation";
+
+const router = express.Router();
+
+const getGameRanks = async (id: number) => {
+  try {
+    const result = await pool.query("SELECT * FROM games WHERE game_id = $1", [
+      id,
+    ]);
+    return result.rows[0].rank_types;
+  } catch (error) {
+    logger.error(error);
+  }
+};
+
+const getAverageRank = async (ranks: Rank[]) => {
+  let average = 0;
+  if (ranks[0].numeric_rank) {
+    ranks.forEach((rank) => (average += rank.numeric_rank));
+    return average / ranks.length;
+  } else {
+    const gameRanks = await getGameRanks(ranks[0].game_id);
+    ranks.forEach((rank) => {
+      average += indexOf(gameRanks, rank.text_rank);
+    });
+    return gameRanks[average / ranks.length];
+  }
+};
+
+/**
+ * @swagger
+ * /ranks:
+ *   get:
+ *     summary: Get all ranks
+ *     tags: [Ranks]
+ *     description: Retrieve all ranks of all users from the database.
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+router.get("/", async (req: Request, res: Response) => {
+  try {
+    const result = await pool.query("SELECT * FROM user_ranks");
+    res.json(result.rows);
+  } catch (error) {
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+/**
+ * @swagger
+ * /ranks/user:
+ *   get:
+ *     summary: Get all ranks of a user on a game
+ *     tags: [Ranks]
+ *     description: Retrieve all ranks for a specific user on a specific game from the database.
+ *     parameters:
+ *       - in: query
+ *         name: userId
+ *         required: true
+ *         description: ID of the user to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: gameId
+ *         required: true
+ *         description: ID of the game to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+
+router.get("/user", async (req: Request, res: Response) => {
```

Comment:
Missed this one - after reading REST conventions, this should probably be `GET /users/{user_id}/games/{game_id}/ranks`